sway-core/src/type_system/unify/unifier.rs

```diff
@@ -181,14 +183,16 @@ impl<'a> Unifier<'a> {
             {
                 self.replace_expected_with_received(expected, r, span)
             }
+
+            // Never type coerces to any other type.
+            // This should be after the unification of self types.
+            (Never, _) => {}
+            (_, Never) => {}
```

Comment:
Is this valid? `Never` should coerce into any type, but no other type should coerce into `Never` except `Never`, or? I mean:
```Rust
let _: u8 = __revert();      // Ok.  Never -> u8.
let _: u8 = { return 123 };  // Ok.  Never -> u8.
let _: ! = __revert();       // Ok.  Never -> Never.
let _: ! = { return 123 };   // Ok.  Never -> u8.

let _: ! = 123u8;  // ERROR.  u8 -> Never.
```